query-core/src/main/java/org/infinispan/query/core/impl/BaseEmbeddedQuery.java

```diff
@@ -82,19 +86,15 @@ public void resetQuery() {
          } else {
             Comparator<Comparable<FilterResult>[]> comparator = getComparator();
             if (comparator == null) {
-               results = StreamSupport.stream(spliterator(), false).collect(Collectors.toList());
+               results = StreamSupport.stream(spliterator(), false).collect(new TimedCollector<>(Collectors.toList(), timeout));
```

Comment:
So the timeout is detected when collecting. But what if the splitterator blocks indefinitely when trying to fetch next?

Reply:
It will probably timeout after the RpcTimeout

Reply:
I could also set the timeout in the stream itself, but only distributed streams support timeout

Reply:
changed